### PR TITLE
'amd' missing from error message for error TS1323

### LINKED ISSUE #33762: "amd" missing from error message for error TS1323
**TypeScript Version:**  3.7.0-dev.20191002

**Search Terms:**
amd, dynamic import, error message

**Code**

```ts
import("./mymodule.js");
```

"module" set to "es2015" in `tsconfig.json`.

**Expected behavior:**

`error TS1323: Dynamic import is only supported when '--module' flag is 'amd' or  'commonjs' or 'esNext' or 'umd'.`

**Actual behavior:**

`error TS1323: Dynamic import is only supported when '--module' flag is 'commonjs' or 'esNext'.`

**Playground Link:** 

**Related Issues:**

### PR DESCRIPTION
**Pull request by a-tarasyuk**
Fixes #33762

cc @DanielRosenwasser

### CODE REVIEW

**Comment by DanielRosenwasser** [comment]
Looks like tests are failing because

> ```
> ERROR: /home/vsts/work/1/s/src/testRunner/parallel/host.ts:150:54 @typescript-eslint/no-unnecessary-type-assertion This assertion is unnecessary since it does not change the type of the expression.
> ERROR: /home/vsts/work/1/s/src/testRunner/parallel/host.ts:159:62 @typescript-eslint/no-unnecessary-type-assertion This assertion is unnecessary since it does not change the type of the expression.
> ```

**Comment by a-tarasyuk** [comment]
@DanielRosenwasser 

> Looks like tests are failing because

Yes, my PR (https://github.com/microsoft/TypeScript/pull/33804) which fix these errors has been merged., just need to restart tests

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -35926,7 +35926,7 @@ namespace ts {
 
         function checkGrammarImportCallExpression(node: ImportCall): boolean {
             if (moduleKind === ModuleKind.ES2015) {
```
[the remark this answers is not in the document]
Comment (a-tarasyuk) [reply src/compiler/checker.ts]: @DanielRosenwasser Do you mean `None` option? I didn't change the condition., if you provide more detailed information about this case, I will add.

**Comment by a-tarasyuk** [comment]
@DanielRosenwasser tests passed ;)

**Pull request merged**